src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -508,6 +486,7 @@ public sealed override bool Equals(Symbol obj, TypeCompareKind compareKind)
             var symbol = obj as SourceLocalSymbol;
             return (object)symbol != null
                 && symbol._identifierToken.Equals(_identifierToken)
+                && symbol.TypeWithAnnotations.Equals(TypeWithAnnotations, compareKind)
```

Comment:
>&& symbol.TypeWithAnnotations.Equals(TypeWithAnnotations, compareKind) [](start = 16, length = 70)

When does this make a difference?

Reply:
It made a difference when we were updating SourceLocalSymbols. It no longer does.

---
In reply to: [329718695](https://github.com/dotnet/roslyn/pull/38769#discussion_r329718695) [](ancestors = 329718695)